### DISCUSSION TITLE
What are TRelatedModel and TChildModel for BelongsTo?

### DISCUSSION THREAD

**Discussion by piljac1** [larastan/larastan#1330] [Q&A] [answered]
I'm gradually implementing Larastan in a project for the first time and when hitting level 6, I started getting errors as such:

> Method App\Models\Session::subscriber() return type with generic class Illuminate\Database\Eloquent\Relations\BelongsTo does not specify its types: TRelatedModel, TChildModel

The related code is the following:

```php
/**
 * Gets the subscriber related to the session.
 * 
 * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
 */
public function subscriber(): BelongsTo
{
    return $this->belongsTo(Subscriber::class);
}
```

Now I understand I have to follow this format:
```php
/**
 * @return \Illuminate\Database\Eloquent\Relations\BelongsTo<\App\Models\SomeModel, \App\Models\SomeModel>
 */
```

But I don't seem to grasp which models should be considered as the related model and the child model.

Thanks in advance for the help :)

**Comment by szepeviktor** [comment, marked as answer]
Hello @piljac1! 👋🏻 
Thank you for starting this discussion.

TRelatedModel: Subscriber
TChildModel: Session

:) or the opposite, I'm not a developer

**Reply by szepeviktor** [reply]
You can always search for the relation: https://github.com/nunomaduro/larastan/search?q=BelongsTo
and look at tests.

**Reply by piljac1** [reply]
Thanks for your answer ! I know it is either Subscriber, Session or Session, Subscriber. But you don't seem 100% sure or is it sarcasm? 🤔

**Reply by piljac1** [reply]
Thanks I'll look at the given tests!

**Reply by szepeviktor** [reply]
> But you don't seem 100% sure or is it sarcasm?

1. I'm not sure at all as I do not write code daily.
2. 💡 It is 0% sarcasm.

**Reply by szepeviktor** [reply]
It seems `BelongsTo<the related model, the current model>` for a simple `BelongsTo` relations.
There are special cases.

**Reply by piljac1** [reply]
From my understanding reading the given tests:
TRelatedModel: Model referred to by the foreign key
TChildModel: Model that has the foreign key (i.e. subscriber_id)
